src/desdemona/plugins/swift.clj

```diff
@@ -0,0 +1,127 @@
+  passed as a request header. If the container already exists, nothing
+  happens."
+  [cf-url auth-token container-name]
+  (let [url (str cf-url "/" container-name)
+        response (http/put url
+                           {:accept :json
+                            :throw-entire-message true
+                            :headers {"X-Auth-Token" auth-token}})]
+    (:status response)))
+
+(defn write-file
+  "Encode the given contents to JSON and write them to the given file on
+  Cloud Files, at the given URL. This requires an auth token, which is passed
+  as a request header. If the file already exists, it is overwritten."
+  [cf-url auth-token container-name file-name contents]
+  (let [url (str cf-url "/" container-name "/" file-name)
+        body (json/generate-string contents)
+        response (http/put url
+                           {:body body
+                            :content-type :json
+                            :throw-entire-message true
+                            :headers {"X-Auth-Token" auth-token}})]
+    (:status response)))
+
+(defn calculate-container-name
+  "We create one container per day, named based on the date. This function
+  generates a container name indicating today's date."
+  []
+  (f/unparse (f/formatter "yyyy-MM-dd") (t/now)))
+
+(defn calculate-file-name
+  "We create one file per batch, named based on the current time. The
+  resolution is intended to be small enough such that each filename will be
+  unique per batch."
+  []
+  (f/unparse (f/formatter "HH:mm:ss-SSS") (t/now)))
+
+(defrecord SwiftWriteRows
+           [auth-url username api-key]
+  p-ext/Pipeline
+
+  (read-batch
+    [_ event]
+    (function/read-batch event))
+
+  (write-batch
+    [_ {:keys [onyx.core/results]}]
+    (let [segments (map :message (mapcat :leaves (:tree results)))]
+      (when (not-empty segments)
+        (let [auth-response (authenticate auth-url username api-key)
+              auth-token (-> auth-response get-token :id)
+              cloud-files-url (-> auth-response get-cloud-files :endpoints
+                                  first :public-url)
+              container-name (calculate-container-name)
+              file-name (calculate-file-name)]
+          (info "Writing" (count segments) "segments to"
+                container-name file-name)
+          (create-container cloud-files-url auth-token container-name)
+          (write-file cloud-files-url auth-token container-name file-name
+                      segments))))
+    {:onyx.core/written? true})
+
+  (seal-resource
+    [_ _]
+    {}))
+
+(defn write-batch
+  "This is the function that gets added to the catalog, and sets up the plugin
+  that actually writes batches to Cloud Files."
+  [pipeline-data]
+  (let [task-map (:onyx.core/task-map pipeline-data)
+        auth-url (:swift/auth-url task-map)
+        username (:swift/username task-map)
+        api-key (:swift/api-key task-map)]
```

Comment:
did destructuring not work out well for this? (I'd even consider juxt, but I'm an animal)